assignments/arrays.js

```diff
@@ -63,34 +75,123 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
 
+// With for loop:
+for (i = 0; i < inventory.length; i++) {
+  if (i === 33) {
+    console.log(
+      `Car 33 is a ${inventory[i].car_year} ${inventory[i].car_make} ${
+        inventory[i].car_model
+      }.`
+    );
+  }
+}
 
+// Without for loop:
+console.log(
+  `Car 33 is a ${inventory[33].car_year} ${inventory[33].car_make} ${
+    inventory[33].car_model
+  }.`
+);
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
+
+// With for loop:
 let lastCar = 0;
-console.log();
+for (i = 0; i < inventory.length; i++) {
+  if (i === inventory.length - 1) {
+    lastCar = inventory[i];
+  }
+}
+console.log(`The last car is a ${lastCar.car_make} ${lastCar.car_model}.`);
+
+// Without for loop:
+console.log(
+  `The last car is a ${inventory[inventory.length - 1].car_make} ${
+    inventory[inventory.length - 1].car_model
+  }.`
+);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
-let carModels = [];
-console.log();
+
+// With for loop:
+// I came up with the general idea here without a lot of trouble. It was really hard to debug, though. I finally figured out to add a break after an item is added to the carModelsWithForLoop array. It was a fun puzzle.
+const carModelsWithForLoop = [];
+for (i = 0; i < inventory.length; i++) {
+  if (i === 0) {
+    carModelsWithForLoop.push(inventory[i].car_model);
+  } else {
+    for (j = 0; j < carModelsWithForLoop.length; j++) {
+      if (
+        inventory[i].car_model < carModelsWithForLoop[j] ||
+        inventory[i].car_model === carModelsWithForLoop[j]
+      ) {
+        carModelsWithForLoop.splice(j, 0, inventory[i].car_model);
+        break;
+      } else {
+        if (j === carModelsWithForLoop.length - 1) {
+          carModelsWithForLoop.push(inventory[i].car_model);
+          break;
+        }
+      }
+    }
+  }
+}
+console.log(carModelsWithForLoop);
```

Comment:
I'm impressed you were able to get such an involved approach implemented properly. I think the original solution didn't actually prohibit the use of `.sort()`, so you could also have just pushed all model names into an array with a single for loop and then apply `.sort()` on it as well